src/components/ForceActionRedirector.js

```diff
@@ -1,24 +1,59 @@
-import { useEffect } from 'react'
+import React, { useEffect } from 'react'
 import { push } from 'react-router-redux'
 import { useUser } from '../stores/user'
 import { store } from '../redux'
 import { PATHS } from '../paths'
+import { useSocket } from '../utils/socketHooks'
+import { useUserSubscriptionStatus } from '../stores/user/subscriptions'
+import TabLimitModal from './TabLimitModal'
 
 const ignoredPages = ['/privacy-policy', '/roadmap']
+const LIMIT_TAB_PAGES = ['/screener', '/watchlists', '/charts']
+const TRY_WAIT_TIME_MS = 3000
+const MAX_TABS_FREE = 2
+const MAX_TABS_PRO = 4
+
+const shouldCheckPage = pathname =>
+  LIMIT_TAB_PAGES.find(page => pathname.startsWith(page))
 
 const ForceActionRedirector = ({ pathname }) => {
   const { user } = useUser()
+  const { isPro, isProPlus } = useUserSubscriptionStatus()
+  const { socket, showTabLimitModal, setShowTabLimitModal } = useSocket()
+  const MAX_TABS = isPro ? MAX_TABS_PRO : MAX_TABS_FREE
+
+  function checkOpenTabs () {
+    if (!socket || isProPlus || !shouldCheckPage(pathname)) {
+      setShowTabLimitModal(false)
+      return
+    }
+
+    const channel = socket.channel(`open_restricted_tabs:${user.id}`, {})
+    channel.join().receive('ok', () => {
+      channel
+        .push('open_restricted_tabs', {}, 10000)
```

Comment:
Better to store `10000` in the variable, otherwise it's hard to understand what it's used for.